src/resdk/query.py

```diff
@@ -380,3 +388,52 @@ def iterate(self, chunk_size=100, show_progress=False):
                     min_id = obj.id
                     pbar.update(1)
                     yield obj
+
+
+class AnnotationFieldQuery(ResolweQuery):
+    """Add additional method to the annotation field query."""
+
+    def from_path(self, full_path: str) -> "AnnotationField":
+        """Get the AnnotationField from full path.
+
+        :raises LookupError: when field at the specified path does not exist.
+        """
+        group_name, field_name = full_path.split(".", maxsplit=1)
+        return self.get(name=field_name, group__name=group_name)
+
+
+class AnnotationValueQuery(ResolweQuery):
+    """Populate Annotation fields with a single query."""
+
+    def _fetch(self):
+        """Make request to the server and populate cache.
+
+        Fetch all values and their fields with 2 queries.
+        """
+        # Execute the query in a single request.
+        super()._fetch()
+
+        missing = {
+            value.field_id: value for value in self._cache if value._field is None
+        }
+        if missing:
+            # Get corresponding annotation field details in a single query and attach it to
+            # the values.
+            for field in self.resolwe.annotation_field.filter(id__in=missing.keys()):
+                missing[field.id]._field = field
+                missing[field.id]._original_values["field"] = field._original_values
+
+    def from_path(self, full_path: str) -> "AnnotationValue":
+        """Get the AnnotationValue from full path.
+
+        :raises LookupError: when field at the specified path does not exist.
+        """
+        group_name, field_name = full_path.split(".", maxsplit=1)
+        return self.get(field__name=field_name, field__group__name=group_name)
```

Comment:
This will probably return all AnnotationValues with given field. So it is probably `self.filter(...)` instead of `self.get(...)`

Reply:
I see, the backend refuses to provide result unless one entity filter is provided. However, my comment is still relevant, consider this filter:
`res.annotation_value.filter(entity__collection=<collection>).from_path("general.species")`
This will result in many results and `AnnotationValueQuery.get` fails since it got multiple results.

Please add a comment in docstring that indicates that user needs to provide a sample (entity) filter.